live_pipeline/extract/extract.py

```diff
@@ -0,0 +1,61 @@
+"""
+Extract the plant data from the api into a dictionary.
+Exports found data to Json is there to make it easy to view but can be removed.
+Exports missing data to json for logging purposes. (Useful to keep)
+"""
+from datetime import datetime
+import json
+import time
+import requests
+
+
+def log_missing(plant_id: int, missing_plants: dict) -> None:
+    """Logs the missing plant id and the current datetime to a json file."""
+
+    current_datetime = datetime.now()
+    missing_plants[str(current_datetime)
+                   ] = f"Data not found for plant_id {plant_id}"
+
+    with open("missing_plants", 'w') as file:
+        json.dump(missing_plants, file, indent=4)
+
+
+class APIError(Exception):
+    """Describes an error triggered by a failing API call."""
+
+    def __init__(self, message: str, code: int = 500):
+        """Creates a new APIError instance."""
+        self.message = message
+        self.code = code
+
+
+def get_plants() -> dict:
+    """ Hit's each endpoint and fetches all the available plant data"""
+    plants = {}
+    missing_plants = {}
+    try:
+        for plant_id in range(1, 51):
```

Comment:
The values `1` and `51` could be supplied as global variables making the script more maintainable if the number of plants was ever varied